src/server/grpc_service.rs

```diff
@@ -777,14 +777,8 @@ impl<T: RaftStoreRouter + 'static> tikvpb_grpc::Tikv for Service<T> {
             stream
                 .map_err(Error::from)
                 .for_each(move |msg| {
-                    let res = match msg.get_region_id() {
-                        0 => Ok(()),
-                        _ => {
-                            RAFT_MESSAGE_RECV_COUNTER.inc();
-                            ch.send_raft_msg(msg)
-                        }
-                    };
-                    future::result(res).map_err(Error::from)
+                    RAFT_MESSAGE_RECV_COUNTER.inc();
+                    future::result(ch.send_raft_msg(msg)).map_err(Error::from)
```

Comment:
> We will not allow region 0 later. So here we need to send an invalid region.

https://github.com/pingcap/tikv/pull/2244#discussion_r137999807

Base on the comment, it should either return an error or end the stream if region id is 0, but here it just ignore.

Reply:
Because we will check the region in raftstore.
Here using 0 is for gRPC buffer hint before, but when we support raftclient flush, we don't need this now.

Reply:
> We will not allow region 0 later.

Do you mean server should reject invalid requests?

> So here we need to send an invalid region.

Do you mean add a new test for ensuring server rejects invalid requests?
The test already sends invalid requests.  0 is the default value.
https://github.com/pingcap/tikv/pull/2244/files#diff-1f7f5096d9c5dc14c96baa97db5e97b8R456

> Because we will check the region in raftstore.

Checks in raftstore will not close the streaming or send an error, so that client does not know if the request, it sends, is vaild or not.

Reply:
no need check here now. we can't know the message is valid here.